[OPAL-930] create repodata.json for every subdir

RR-USAF opened the pull request:
correctly create repodata.json for every subir in the solution.

RR-USAF committed:
[OPAL-930] create repodata.json for every subdir

JR-USAF reviewed (approved):
I didn't see anything that stuck out to me and all tests passed.

![image](https://user-images.githubusercontent.com/117106269/212393163-4d644b7d-06fc-4877-93ea-775e2dd3ef87.png)

RR-USAF committed:
fix: fixed bug in filtering channel on Windows

- added gitignore rules
- fixed bug vendoring from windows
- changed the logic of when to add virtual packages to the spec list

RR-USAF committed:
fix: updated documentation

IM-USAF commented on conda_vendor/conda_vendor.py:
```diff
@@ -70,45 +75,33 @@
 
-def _blue(s, bold=True):
-    click.echo(click.style(s, fg="blue", bg="black", bold=bold))
+def _blue(msg: str, bold: bool = True):
```
Curious why you didn't break out some of this house-keeping stuff?

RR-USAF replied:
i'm not sure what you mean, what have you got in mind?  I have no strong opinions about this.

IM-USAF replied:
Maybe put it in a `_terminal.py` or `_utils.py` or something.

RR-USAF committed:
fix: bugs in tests because of new virtual package handling

RR-USAF committed:
fix: typo

IM-USAF commented on conda_vendor/conda_vendor.py:
```diff
@@ -123,19 +116,31 @@ def _get_environment_name(environment_file: Path) -> str:
 
 
 def _get_virtual_packages(
```
conda_vendor.py is pretty big. May help other contributors if you separate out some stuff, such as virtual package-related funcs.

RR-USAF replied:
i want to refactor more completely in the future, to separate the library stuff from the ui stuff. 

the counter argument is that it's ~800 lines with comments, ~600 lines of code (with at least 200 lines just doing ui), all in all, not too big. 

i'm not sure how much easier it is to understand if we just hide stuff away, and all the code kind of depends on the other code there, i couldn't find good module separation, yet.  

right now, all of conda vendor sits in conda_vendor.py which means there are no secrets.

IM-USAF replied:
I think refactoring can wait.

IM-USAF commented on conda_vendor/conda_vendor.py:
```diff
@@ -470,9 +506,9 @@ def yaml_dump_ironbank_manifest(package_list: List[FetchAction]):
         resources["resources"].append(resource)
 
-    yaml = YAML()
+    yaml_ = YAML()
```
Is there some semantic reason for the trailing underscore?

RR-USAF replied:
to not shadow import yaml at the top.  it was to silence a pylint warning.

IM-USAF replied:
makes sense
